test/unit/ngsiv2/ngsiService/subscriptions-test.js

```diff
@@ -0,0 +1,351 @@
+                .reply(201, null, {'Location': '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8'});
+
+            iotAgentLib.clearAll(function() {
+                request(optionsProvision, function(error, result, body) {
+                    done();
+                });
+            });
+        });
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+        iotAgentLib.setNotificationHandler();
+        iotAgentLib.clearAll(function() {
+            iotAgentLib.deactivate(done);
+        });
+    });
+
+    describe('When a client invokes the subscribe() function for device', function() {
+        it('should send the appropriate request to the Context Broker', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    should.not.exist(error);
+
+                    contextBrokerMock.done();
+
+                    done();
+                });
+            });
+        });
+        it('should store the subscription ID in the Device Registry', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                        should.not.exist(error);
+                        should.exist(device);
+                        should.exist(device.subscriptions);
+                        device.subscriptions.length.should.equal(1);
+                        device.subscriptions[0].id.should.equal('51c0ac9ed714fb3b37d7d5a8');
+                        device.subscriptions[0].triggers[0].should.equal('attr_name');
+                        done();
+                    });
+                });
+            });
+        });
+    });
+    describe('When a client invokes the unsubscribe() function for an entity', function() {
+        beforeEach(function(done) {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .delete('/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8')
+                .reply(204);
+
+            done();
+        });
+        it('should delete the subscription from the CB', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    iotAgentLib.unsubscribe(device, '51c0ac9ed714fb3b37d7d5a8', function(error) {
+                        iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+
+                            contextBrokerMock.done();
```

Comment:
In this case, `DELETE /v2/subscription/<id>` should be invoked on the CB. Looking to the mock interceptors (the so discussed code in lines 67-101 :) I don't see any .delete() interceptor being declared. Thus, how do you check that the subscription is actually deleted at CB in this test?

Reply:
In fact it seems that in other test this is correctly done, see https://github.com/dcalvoalonso/iotagent-node-lib/pull/4/files#diff-665dfc645ea8386f7fc737fc0476026fR191.

Reply:
The .delete interceptor is defined in https://github.com/dcalvoalonso/iotagent-node-lib/blob/task/bidirectionalityPlugin/test/unit/ngsiv2/ngsiService/subscriptions-test.js#L151

Reply:
NTC

I was looking the initial declaration of the mock and miss that one, sorry... :S